src/plugins/intel_gpu/src/runtime/engine.cpp

```diff
@@ -211,14 +211,13 @@ uint64_t engine::get_used_device_memory(allocation_type type) const {
     return _memory_usage_data[static_cast<size_t>(type)].load();
 }
 
-std::map<std::string, uint64_t> engine::get_memory_statistics() const {
-    std::map<std::string, uint64_t> statistics;
+std::map<std::string, uint64_t> engine::get_memory_statistics() {
     const auto add_stat = [&](allocation_type type) {
         auto idx = static_cast<size_t>(type);
         auto value = _memory_usage_data[idx].load();
-        if (value != 0) {
-            std::ostringstream oss;
-            oss << type;
+        std::ostringstream oss;
+        oss << type;
+        if (value != 0 || statistics.count(oss.str())) {
             statistics[oss.str()] = value;
         }
```

Comment:
We can just ignore `value != 0` condition here, and keep all other parts unchanged
```suggestion
        std::ostringstream oss;
        oss << type;
        statistics[oss.str()] = value;
```